### PR TITLE
chore: refactor the structure of logging

### PR DESCRIPTION
**Pull request by rudeigerc**
Signed-off-by: Yuchen Cheng <rudeigerc@gmail.com>

### Ⅰ. Describe what this PR does

This pull request refactors the structure of logging according to the [documentation](https://github.com/koordinator-sh/koordinator/blob/main/docs/contributors/logging.md).

### Ⅱ. Does this pull request fix one issue?

Related to #222 

### Ⅲ. Describe how to verify it

N/A

### Ⅳ. Special notes for reviews

I follow some practical rules to decide which logging level the log should be. Please feel free to leave reviews if you consider that they are not suitable.

- `klog.V(1)`
  - The starting and shutting down of the main components.
  - Information related to configurations.
- `klog.V(2)`
  - The changing of state.
- `klog.V(4)`
  - The logging of values for debugging.
- `klog.V(5)`
  - Some tracing info.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Comment by hormes** [comment]
nice work, but the biggest problem at the moment is that error/warning/info is not very well differentiated.

File: cmd/koord-runtime-proxy/main.go [hunk 1]
```diff
@@ -64,5 +64,5 @@ func main() {
 
 	stopCh := genericapiserver.SetupSignalHandler()
 	<-stopCh
-	klog.Info("koordiantor runtime-proxy shutting down")
+	klog.V(1).Info("koordiantor runtime-proxy shutting down")
```
Comment (hormes) [inline cmd/koord-runtime-proxy/main.go:67]: Compared with V1, the direct Info is more concise. It seems that the distinction between the two is not that big.

File: cmd/koord-scheduler/app/server.go [hunk 2]
```diff
@@ -207,7 +207,7 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 				select {
 				case <-ctx.Done():
 					// We were asked to terminate. Exit 0.
-					klog.Info("Requested to terminate. Exiting.")
+					klog.V(2).Info("Requested to terminate. Exiting.")
```
Comment (hormes) [inline cmd/koord-scheduler/app/server.go:210]: Info is better here

File: cmd/koordlet/main.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func main() {
 	}
 
 	// Get a config to talk to the apiserver
-	klog.Info("Setting up client for koordlet")
+	klog.V(1).Info("Setting up client for koordlet")
```
Comment (hormes) [inline cmd/koordlet/main.go:64]: Info is better here

File: cmd/koordlet/main.go [hunk 4]
```diff
@@ -76,7 +76,7 @@ func main() {
 
 	// Expose the Prometheus http endpoint
 	go func() {
-		klog.Infof("Starting prometheus server on %v", *options.ServerAddr)
+		klog.V(1).Infof("Starting prometheus server on %v", *options.ServerAddr)
```
Comment (hormes) [inline cmd/koordlet/main.go:79]: Info is simpler here

File: cmd/koordlet/main.go [hunk 5]
```diff
@@ -86,6 +86,6 @@ func main() {
 	}()
 
 	// Start the Cmd
-	klog.Info("Starting the koordlet daemon")
+	klog.V(1).Info("Starting the koordlet daemon")
```
Comment (hormes) [inline cmd/koordlet/main.go:89]: Info is simpler here

File: pkg/koordlet/audit/auditor.go [hunk 6]
```diff
@@ -132,7 +132,7 @@ func (a *auditor) HttpHandler() func(http.ResponseWriter, *http.Request) {
 		sizeStr := r.URL.Query().Get("size")
 		pageToken := r.URL.Query().Get("pageToken")
 
-		klog.Infof("handle query client=%v pageToken=%v size=%v", r.RemoteAddr, pageToken, sizeStr)
+		klog.V(4).InfoS("handle query", "client", r.RemoteAddr, "pageToken", pageToken, "size", sizeStr)
```
Comment (hormes) [inline pkg/koordlet/audit/auditor.go:135]: <= V(3) is better here, we want to log who visited here

File: pkg/koordlet/audit/event_logger.go [hunk 7]
```diff
@@ -129,7 +129,7 @@ func (e *eventWriter) openNewLogWriter() error {
 			klog.Errorf("failed to list files %v", pattern)
 		} else {
 			for i := 0; i < len(files)-e.maxFileNum+1; i++ {
-				klog.Infof("clean audit file %v", files[i])
+				klog.V(4).Infof("clean audit file %v", files[i])
```
Comment (hormes) [inline pkg/koordlet/audit/event_logger.go:132]: Info is better here, we want to log who visited here

File: pkg/koordlet/koordlet.go [hunk 8]
```diff
@@ -70,11 +70,11 @@ func NewDaemon(config *config.Configuration) (Daemon, error) {
...
 		return nil, fmt.Errorf("failed to new daemon: NODE_NAME env is empty")
 	}
-	klog.Infof("NODE_NAME is %v,start time %v", nodeName, float64(time.Now().Unix()))
+
+	klog.V(1).InfoS("Starting daemon", "nodeName", nodeName, "start time", float64(time.Now().Unix()))
 	metrics.RecordKoordletStartTime(nodeName, float64(time.Now().Unix()))
 
-	klog.Infof("sysconf: %+v,agentMode:%v", system.Conf, system.AgentMode)
-	klog.Infof("kernel version INFO : %+v", system.HostSystemInfo)
+	klog.V(1).InfoS("Starting daemon", "sysconf", system.Conf, "agentMode", system.AgentMode, "kernel version", system.HostSystemInfo)
```
Comment (hormes) [inline pkg/koordlet/koordlet.go:77]: Info is simpler here

File: pkg/koordlet/koordlet.go [hunk 9]
```diff
@@ -70,11 +70,11 @@ func NewDaemon(config *config.Configuration) (Daemon, error) {
 	if len(nodeName) == 0 {
 		return nil, fmt.Errorf("failed to new daemon: NODE_NAME env is empty")
 	}
-	klog.Infof("NODE_NAME is %v,start time %v", nodeName, float64(time.Now().Unix()))
+
+	klog.V(1).InfoS("Starting daemon", "nodeName", nodeName, "start time", float64(time.Now().Unix()))
```
Comment (hormes) [inline pkg/koordlet/koordlet.go:74]: Info is simpler here

File: pkg/koordlet/koordlet.go [hunk 10]
```diff
@@ -193,7 +193,7 @@ func (d *daemon) Run(stopCh <-chan struct{}) {
 		}
 	}()
 
-	klog.Info("Start daemon successfully")
+	klog.V(1).Info("Start daemon successfully")
```
Comment (hormes) [inline pkg/koordlet/koordlet.go:196]: Info is simpler here

**Comment by hormes** [comment]
There are too many comments above, the suggestion is this
1. Roll back all V(1) changes and keep Info, which is easier
2. First pay attention to the level >= 4, which is the debug log
3. V(2) V(3) for further discussion.

**Comment by hormes** [comment]
@rudeigerc

**Comment by rudeigerc** [comment]
@hormes Thanks for your review! :) I will make an update ASAP.

**Comment by jasonliu747** [comment]
Hi @rudeigerc , any progress on this PR?

**Pull request closed without merging**